ldap/src/main/java/pl/edu/icm/unity/ldap/endpoint/LdapApacheDSInterceptor.java

```diff
@@ -627,19 +689,182 @@ private EntryFilteringCursor getUnityUser(SearchOperationContext searchContext,
+                );
+                for (String at : default_attributes.split(","))
+                {
+                    attributes.add(
+                        new AttributeTypeOptions(
+                            schemaManager.lookupAttributeTypeRegistry(at.trim())
+                        )
+                    );
+                }
+            }
+
+            // iterate through requested attributes and try to match them with
+            // identity attributes
+            String[] aliases = configuration.getValue(
+                LdapServerProperties.USER_NAME_ALIASES
+            ).split(",");
+            for (String alias : aliases)
+            {
+                for (AttributeTypeOptions ao : attributes)
+                {
+                    String aName = ao.getAttributeType().getName();
+                    if (aName.equals(alias))
+                    {
+                        String requestDn = String.format("%s=%s", alias, username);
+                        // FIXME: is this what we want in all cases?
+                        if (null != searchContext.getDn())
+                        {
+                            requestDn += "," + searchContext.getDn().toString();
+                        }
+                        entry.setDn(requestDn);
+                    } else {
+                        addAttribute(aName, userEntity, username, attrs, entry);
+                    }
+                }
+            }
+
+            // the purpose of a search can be to get DN (from another attribute)
+            //  - even if no attributes are in getReturningAttributes()
+            if (null == entry.getDn() || entry.getDn().isEmpty() ) {
+                entry.setDn(searchContext.getDn());
+            }
+            */
+            Entry userEntry = getUnityUserEntry(searchContext, username);
+            if (userEntry != null) {
+                return new EntryFilteringCursorImpl(new SingletonCursor<>(userEntry),
+                        searchContext, lsf.getDs().getSchemaManager()
+                );
+            }
+
+        } catch (Exception e)
+        {
+            
             throw new LdapOtherException("Error establishing user information", e);
         }
 
         return emptyResult(searchContext);
     }
 
+    /**
+     * Return an LDAP attribute with its value set to the first value of the 
+     * unity attribute specified by attributeName, or null if the unity attribute 
+     * is not found in the collection.
+     * 
+     * @param unityAttrs Collection of unity attributes
+     * @param attributeName Name of the unity attribute
+     * @param uid LDAP attribute name
+     * @param upId LDAP attribute name (OID)
+     * @return  LDAP attribute
+     * @throws LdapException
+     */
+    private Attribute mapUnityAttributeToLdapAttribute(Collection<AttributeExt<?>> unityAttrs, String attributeName, String uid, String upId) throws LdapException {
```

Comment:
In the current impl there is no problem here, but note that after creating a configurable mapping of attrs (see above comment) this will become quite ineffective: it will be better to first create and then pass a map of attributes by name.